presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java

```diff
@@ -430,12 +444,12 @@ private AggregationNode createNonDistinctAggregation(
                     aggregationOutputSymbolsMapBuilder.put(newVariable, entry.getKey());
                     // Handling for cases when mask symbol appears in non distinct aggregations too
                     // Now the aggregation should happen over the duplicate symbol added before
-                    List<Expression> arguments;
-                    if (!duplicatedDistinctVariable.equals(distinctVariable) && entry.getValue().getArguments().contains(new SymbolReference(distinctVariable.getName()))) {
-                        ImmutableList.Builder<Expression> argumentsBuilder = ImmutableList.builder();
-                        for (Expression argument : aggregation.getArguments()) {
-                            if (new SymbolReference(distinctVariable.getName()).equals(argument)) {
-                                argumentsBuilder.add(new SymbolReference(duplicatedDistinctVariable.getName()));
+                    List<RowExpression> arguments;
+                    if (!duplicatedDistinctVariable.equals(distinctVariable) && extractVariables(entry.getValue().getArguments(), symbolAllocator.getTypes()).contains(distinctVariable)) {
+                        ImmutableList.Builder<RowExpression> argumentsBuilder = ImmutableList.builder();
+                        for (RowExpression argument : aggregation.getArguments()) {
+                            if (getVariable(argument, symbolAllocator.getTypes()).map(a -> a.equals(distinctVariable)).orElse(false)) {
```

Comment:
I would just put the logic of `getVariable` in this loop. `getVariable` is too generic name to make ambiguity.